integration/messaging/messaging_test.go

```diff
@@ -0,0 +1,315 @@
+package messaging
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+
+	"firebase.google.com/go/integration/internal"
+	"firebase.google.com/go/messaging"
+)
+
+var projectID string
+var client *messaging.Client
+
+var testFixtures = struct {
+	token     string
+	topic     string
+	condition string
+}{}
+
+// Enable API before testing
+// https://console.developers.google.com/apis/library/fcm.googleapis.com/?project=
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if testing.Short() {
+		log.Println("skipping Messaging integration tests in short mode.")
+		return
+	}
+
+	token, err := ioutil.ReadFile(internal.Resource("integration_token.txt"))
```

Comment:
All these files require a lot of setting up. Lets keep this simple by doing something like: https://github.com/firebase/firebase-admin-java/blob/a4661a0f44d1dc35d1531eac78f9e3459d6bccb7/src/test/java/com/google/firebase/messaging/FirebaseMessagingIT.java